repairnator/src/main/java/fr/inria/spirals/repairnator/process/step/BuildShouldFail.java

```diff
@@ -0,0 +1,45 @@
+package fr.inria.spirals.repairnator.process.step;
+
+import fr.inria.spirals.jtravis.entities.BuildStatus;
+import fr.inria.spirals.repairnator.process.ProjectState;
+
+/**
+ * Created by fermadeiral.
+ */
+public class BuildShouldFail implements ContractForGatherTestInformation {
+
+    @Override
+    public void makeADecision(GatherTestInformation gatherTestInformation) {
+        if (gatherTestInformation.getState() == ProjectState.HASTESTFAILURE) {
```

Comment:
I found the contract is not clear enough here: basically in method `makeADecision` you can do whatever you want an never take a decision... 

It would be more precise if the method is called "shouldBeStopped" and return a boolean: then we can use it directly as a test method to set shouldStop value. WDYT?